src/java/org/apache/cassandra/index/sai/disk/v1/sortedterms/SortedTermsWriter.java

```diff
@@ -0,0 +1,216 @@
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.index.sai.disk.v1.sortedterms;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.util.Arrays;
+import javax.annotation.Nonnull;
+import javax.annotation.concurrent.NotThreadSafe;
+
+import com.google.common.base.Preconditions;
+
+import org.apache.cassandra.index.sai.disk.io.IndexOutputWriter;
+import org.apache.cassandra.index.sai.disk.v1.MetadataWriter;
+import org.apache.cassandra.index.sai.disk.v1.bitpack.NumericValuesWriter;
+import org.apache.cassandra.index.sai.utils.SAICodecUtils;
+import org.apache.cassandra.io.tries.IncrementalDeepTrieWriterPageAware;
+import org.apache.cassandra.io.util.FileUtils;
+import org.apache.cassandra.utils.bytecomparable.ByteComparable;
+import org.apache.cassandra.utils.bytecomparable.ByteSource;
+import org.apache.lucene.store.IndexOutput;
+import org.apache.lucene.util.BytesRef;
+import org.apache.lucene.util.BytesRefBuilder;
+import org.apache.lucene.util.StringHelper;
+
+import static org.apache.cassandra.index.sai.disk.v1.trie.TrieTermsDictionaryReader.trieSerializer;
+
+/**
+ * Writes an ordered sequence of terms for use with {@link SortedTermsReader}.
+ * <p>
+ * Terms must be added in lexicographical ascending order.
+ * Terms can be of varying lengths.
+ *
+ * <p>
+ * Important implementation note: SAI blocked packed readers are slow,
+ * and Lucene MonotonicBlockPackedReader is slow.  Using them
+ * will cause this class to slow considerably.
+ *
+ * For documentation of the underlying on-disk data structures, see the package documentation.
+ *
+ * @see SortedTermsReader
+ * @see org.apache.cassandra.index.sai.disk.v1.sortedterms
+ */
+@NotThreadSafe
+public class SortedTermsWriter implements Closeable
+{
+    // The TERMS_DICT_ constants allow for quickly determining the id of the current block based on a point id
+    // or to check if we are exactly at the beginning of the block.
+    // Terms data are organized in blocks of (2 ^ TERMS_DICT_BLOCK_SHIFT) terms.
+    // The blocks should not be too small because they allow prefix compression of
+    // the terms except the first term in a block.
+    // The blocks should not be too large because we can't just ranfomly jump to the term inside the block,
+    // but we have to iterate through all the terms from the start of the block.
```

Comment:
What if we just move this comment block to the class JavaDoc, where we can reference them w/, for example, `{@link #TERMS_DICT_BLOCK_SHIFT}`?